infracost register outputs lines twice on Windows

I noticed the name/email output was doubled:
```
C:\infracost>infracost.exe register
Please enter your name and email address to get an API key.
See our FAQ (https://www.infracost.io/docs/faq) for more details.
Name: Txxxxr
Name: Txxxxr
Email: xxxx@gmail.com
Email: xxxx@gmail.com

Thank you Txxxxxr!
Your API key is: xxxx
```

@tim775 

I am getting this error when i run `make windows`
```
process_begin: CreateProcess(NULL, env bash C:\Users\Rupan\ops\infracost\scripts\get_version.sh HEAD, ...) failed.
Makefile:3: pipe: No error
env GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build -ldflags="-X 'github.com/infracost/infracost/internal/version.Version='" -v -o build/infracost.exe github.com/infracost/infracost/cmd/infracost
process_begin: CreateProcess(NULL, env GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build "-ldflags=-X 'github.com/infracost/infracost/internal/version.Version='" -v -o build/infracost.exe github.com/infracost/infracost/cmd/infracost, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [Makefile:25: windows] Error 2
```
Seems like I am unable to run get_version.sh bash file in windows. 
I then ran the bash file using bash and got version as `v0.9.8+33-g2ad297b`. Then I hardcoded verison in the make file to see if it works but I am getting this error. I get not found when I try to curl this url `github.com/infracost/infracost/internal/version.Version=v0.9.8+33-g2ad297b`

```
env GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build -ldflags="-X 'github.com/infracost/infracost/internal/version.Version=v0.9.8+33-g2ad297b'" -v -o build/infracost.exe github.com/infracost/infracost/cmd/infracost
process_begin: CreateProcess(NULL, env GOOS=windows GOARCH=amd64 CGO_ENABLED=0 go build "-ldflags=-X 'github.com/infracost/infracost/internal/version.Version=v0.9.8+33-g2ad297b'" -v -o build/infracost.exe github.com/infracost/infracost/cmd/infracost, ...) failed.
make (e=2): The system cannot find the file specified.
make: *** [Makefile:25: windows] Error 2
```

--- a/cmd/infracost/register.go
+++ b/cmd/infracost/register.go
@@ -4,6 +4,7 @@ import (
 	"fmt"
 	"os"
 	"regexp"
+	"runtime"
 	"strings"
 
 	"github.com/manifoldco/promptui"
@@ -143,88 +144,73 @@ func registerCmd(ctx *config.RunContext) *cobra.Command {
 }
 
 func promptForName() (string, error) {
-	p := promptui.Prompt{
-		Label: "Name",
-		Validate: func(input string) error {
-			input = strings.TrimSpace(input)
-			if input == "" {
-				return errors.New("Please enter a name")
-			}
-			return nil
-		},
-	}
-	name, err := p.Run()
-	name = strings.TrimSpace(name)
+	name, err := stringPrompt("Name", func(input string) error {
+		input = strings.TrimSpace(input)
+		if input == "" {
+			return errors.New("Please enter a name")
+		}
+		return nil
+	})
 
 	return name, err
 }
 
 func promptForCIDocs(isRegenerate bool) (bool, error) {
-	label := "Would you like to see our CI/CD integration docs?"
+	label := "Would you like to see our CI/CD integration docs"
 	if isRegenerate {
-		label = "Do you plan to use this API key in CI?"
-	}
-	p := promptui.Prompt{
-		Label:     label,
-		IsConfirm: true,
+		label = "Do you plan to use this API key in CI"
 	}
 
-	_, err := p.Run()
-	if err != nil {
-		if errors.Is(err, promptui.ErrAbort) {
-			return false, nil
-		}
-		return false, err
-	}
-
-	return true, nil
+	return yesNoPrompt(label)
 }
 
 func promptForEmail() (string, error) {
-	p := promptui.Prompt{
-		Label: "Email",
-		Validate: func(input string) error {
-			input = strings.TrimSpace(input)
-			if input == "" {
-				return errors.New("Please enter an email")
-			}
-			match, err := regexp.MatchString(".+@.+", input)
-			if err != nil {
-				return errors.New("Unable to validate email")
-			}
-			if !match {
-				return errors.New("Please enter a valid email")
-			}
-			return nil
-		},
-	}
-	email, err := p.Run()
-	email = strings.TrimSpace(email)
+	email, err := stringPrompt("Email", func(input string) error {
+		if input == "" {
+			return errors.New("Please enter an email")
+		}
+		match, err := regexp.MatchString(".+@.+", input)
+		if err != nil {
+			return errors.New("Unable to validate email")
+		}
+		if !match {
+			return errors.New("Please enter a valid email")
+		}
+		return nil
+	})
 
 	return email, err
 }
 
 func promptOverwriteAPIKey() (bool, error) {
-	p := promptui.Prompt{
-		Label:     "Would you like to overwrite your existing saved API key",
-		IsConfirm: true,
-	}
-
-	_, err := p.Run()
-	if err != nil {
-		if errors.Is(err, promptui.ErrAbort) {
-			return false, nil
-		}
-
-		return false, err
-	}
-
-	return true, nil
+	return yesNoPrompt("Would you like to overwrite your existing saved API key")
 }
 
 func promptGenerateNewKey() (bool, error) {
+	return yesNoPrompt("Would you like to generate a new API key")
+}
+
+func stringPrompt(label string, validate ui.ValidateFn) (string, error) {
+	if runtime.GOOS == "windows" {
+		return ui.StringPrompt(label, validate), nil
+	}
+
 	p := promptui.Prompt{
-		Label:     "Would you like to generate a new API key",
+		Label:    label,
+		Validate: promptui.ValidateFunc(validate),
+	}
+	name, err := p.Run()
+	name = strings.TrimSpace(name)
+	return name, err
+}
+
+func yesNoPrompt(label string) (bool, error) {
+	if runtime.GOOS == "windows" {
+		return ui.YesNoPrompt(label + "?"), nil
+	}
+
+	p := promptui.Prompt{
+		Label:     label,
 		IsConfirm: true,
 	}
 

--- a/internal/ui/promts.go
+++ b/internal/ui/promts.go
@@ -1,0 +1,63 @@
+package ui
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strings"
+)
+
+// ValidateFn represents a validation function type for prompts. Function of
+// this type should accept a string input and return an error if validation fails;
+// otherwise return nil.
+type ValidateFn func(input string) error
+
+// StringPrompt provides a single line for user input. It accepts an optional
+// validation function.
+func StringPrompt(label string, validate ValidateFn) string {
+	input := ""
+	reader := bufio.NewReader(os.Stdin)
+
+	for {
+		fmt.Fprint(os.Stdout, label+": ")
+		input, _ = reader.ReadString('\n')
+		input = strings.TrimSpace(input)
+
+		if validate == nil {
+			return input
+		}
+
+		err := validate(input)
+		if err == nil {
+			break
+		}
+
+		fmt.Fprintln(os.Stderr, err)
+	}
+
+	return input
+}
+
+// YesNoPrompt provides a yes/no user input. "No" is a default answer if left
+// empty.
+func YesNoPrompt(label string) bool {
+	choices := "y/N"
+	reader := bufio.NewReader(os.Stdin)
+
+	for {
+		fmt.Fprintf(os.Stdout, "%s [%s] ", label, choices)
+		input, _ := reader.ReadString('\n')
+		input = strings.TrimSpace(input)
+
+		if input == "" {
+			return false
+		}
+
+		switch strings.ToLower(input) {
+		case "y", "yes":
+			return true
+		case "n", "no":
+			return false
+		}
+	}
+}

